openedx/adg/lms/applications/analytics.py

```diff
@@ -0,0 +1,191 @@
+
+    Arguments:
+        user (User): User for which to emit the event
+        application_step (str): Step of the application to update the user information for
+    """
+    if application_step == CONTACT_INFO:
+        attributes = {
+            'isSaudi': user.extended_profile.saudi_national,
+            'age': user.extended_profile.age,
+            'heardAboutUs': user.extended_profile.hear_about_omni,
+        }
+
+    elif application_step == EXPERIENCE:
+        first_education = user.application.applications_educations.first()
+        education_start_date, education_end_date = _get_start_date_and_end_date(first_education)
+        attributes = {
+            'school': first_education.name_of_school,
+            'degree': first_education.get_degree_display(),
+            'educationStartDate': education_start_date,
+            'educationEndDate': education_end_date,
+        }
+
+        first_experience = user.application.applications_workexperiences.first()
+        if first_experience:
+            experience_start_date, experience_end_date = _get_start_date_and_end_date(first_experience)
+            attributes.update({
+                'workOrganization': first_experience.name_of_organization,
+                'jobTitle': first_experience.job_position_title,
+                'workStartDate': experience_start_date,
+                'workEndDate': experience_end_date
+            })
+
+    else:
+        attributes = {
+            'businessLine': user.application.business_line.title
+        }
+        emit_application_progress(user)
+
+    segment.identify(user.id, attributes)
+    segment.track(user.id, APPLICATION_EVENTS.get(application_step), attributes)
+
+
+def _get_start_date_and_end_date(position):
+    """
+    Check if the position is ongoing (current), and returns the start_date and the end_date of the
+    position.
+
+    Arguments:
+        position (UserStartAndEndDates): WorkExperience or Education to get the start and end date for
+
+    Returns:
+        (str, str): A tuple containing start_date and end_date in a tuple. If the position is ongoing
+            the end_date is `In Progress`.
+    """
+    format_for_month_and_year_only = '%B, %G'
+    start_date = get_formatted_date(
+        datetime(position.date_started_year, position.date_started_month, 1), format_for_month_and_year_only
+    )
+
+    is_position_current = (
+        (hasattr(position, 'is_current_position') and position.is_current_position) or
+        (hasattr(position, 'is_in_progress') and position.is_in_progress)
+    )
+    if is_position_current:
+        end_date = IN_PROGRESS
+    else:
+        end_date = get_formatted_date(
+            datetime(position.date_completed_year, position.date_completed_month, 1), format_for_month_and_year_only
+        )
+
+    return start_date, end_date
+
+
+def emit_enrollment_event_for_bu_courses(user, status):
```

Comment:
This event is called when bu courses are started and completed therefore we should change the name of the event

Reply:
Got it, renaming it to `emit_event_for_bu_courses_progress`